src/store/mutations.ts

```diff
@@ -0,0 +1,48 @@
+import { type MutationTree } from 'vuex'
+import * as StoreTypes from '@/store/types'
+
+const StoreMutations: MutationTree<StoreTypes.OverallState> = {
+  resetData(state, payload: { users: StoreTypes.BlogUsersState; posts: StoreTypes.BlogPost[] }) {
+    state.users = { ...state.users, ...payload.users }
+    state.posts = payload.posts
+
+    state.reactions = {}
+  },
+  createBlogPost(state: StoreTypes.OverallState, blogPost: StoreTypes.BlogPost) {
+    state.posts = [...state.posts, blogPost]
+  },
+  updateBlogPost(state: StoreTypes.OverallState, blogPost: StoreTypes.BlogPost) {
+    const postIndex = state.posts.findIndex((post) => post.postId === blogPost.postId)
+    if (postIndex >= 0) {
+      state.posts[postIndex] = blogPost
+    }
+  },
```

Comment:
when "mutating the state", should the state be overwritten technically?
For example, `state.posts` is an array of Post Obejcts. So should this be
```
const updatedPosts = [...state.posts]
updatedPosts[postIndex] = blogPost
state.posts = updatedPosts
```

Reply:
And idk what usecase might need this, btu what if I have a mutation that needs to  update two first-level properties of the state,

would it be advisable to do something like this:
```
state = {
  ...state,
  property1: {...state.property1, [newThing]: [...]}
  property2: {...state.property1, [newThing2]: [...]}
```
Or is it all the same if its just:
```
state[property1] = {...state.property1, [newThing]: [...]}
state[property2] = {...state.property2, [newThing2]: [...]}
```

Reply:
It's best practice to separate the mutations instead